### DISCUSSION TITLE
[Popup,Modal] Inline popup doesn't show on content inside modal

### DISCUSSION THREAD

**Discussion by dlynchcodes** [fomantic/Fomantic-UI#1640] [Modules (Modal, Dropdown, Popup, Calendar, Toast, ...)] [answered]
# Bug Report
Inline popups don't display when they are used inside a modal

## Steps to reproduce
1. Add content and an inline popup to a modal
2. Show the modal
3. See the lack of a popup on the content

## Expected result
The popup is displayed

## Actual result
The popup is not displayed

## Testcase
[JSFiddle](https://jsfiddle.net/9p8h2qty/)

## Version
2.8.6

**Comment by lubber-de** [comment, marked as answer]
This is due to the modal being detached from the dom by default, loosing previous attached events.
Workarounds:

1. Use `detachable: false`
https://jsfiddle.net/u3mx7gzj/

2. Instantiate the popup after the modal is instantiated
https://jsfiddle.net/u3mx7gzj/2/

Both jsfiddles are based on your example. However the popup won't be shown in any case when a dropdown is actively opened, which is the case in your example because the modal only contains the dropdown field, which gets focussed, thus opened by default.
To avoid this, use `autofocus:false` to see the popup immediatly
https://jsfiddle.net/u3mx7gzj/3/

![popupmodal](https://user-images.githubusercontent.com/18379884/90047521-537a7a80-dcd2-11ea-8250-f16a69760dc1.gif)

**Comment by dlynchcodes** [comment]
Interesting! I can definitely instantiate the popup when the modal is shown. For my use case I'm already using `autofocus:false` but just didn't use it in the JSFiddle.

What are the ramifications of setting the modal to `detachable:false` ?

**Comment by lubber-de** [comment]
> What are the ramifications of setting the modal to `detachable:false` ?

Existing events (like the popups hover event) will be kept. They get lost when the modal is detached from the DOM

**Comment by dlynchcodes** [comment]
That might make more sense for my use case. I will play around with it. Thanks for clearing up my confusion!

**Comment by ko2in** [comment]
Close since @lubber-de already provided the solution. Please feel free to reopen if it doesn't resolve.
